nifi-toolkit/nifi-toolkit-tls/src/main/java/org/apache/nifi/toolkit/tls/standalone/TlsToolkitStandalone.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.nifi.toolkit.tls.properties.NiFiPropertiesWriterFactory;
+import org.apache.nifi.toolkit.tls.util.OutputStreamFactory;
+import org.apache.nifi.toolkit.tls.util.TlsHelper;
+import org.bouncycastle.openssl.jcajce.JcaMiscPEMGenerator;
+import org.bouncycastle.util.io.pem.PemWriter;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import java.util.List;
+
+public class TlsToolkitStandalone {
+    public static final String NIFI_KEY = "nifi-key";
+    public static final String NIFI_CERT = "nifi-cert";
+    public static final String ROOT_CERT_PRIVATE_KEY = "rootCert.key";
+    public static final String ROOT_CERT_CRT = "rootCert.crt";
+    public static final String NIFI_PROPERTIES = "nifi.properties";
+
+    private final OutputStreamFactory outputStreamFactory;
+
+    public TlsToolkitStandalone() {
+        this(FileOutputStream::new);
+    }
+
+    public TlsToolkitStandalone(OutputStreamFactory outputStreamFactory) {
+        this.outputStreamFactory = outputStreamFactory;
+    }
+
+    public void createNifiKeystoresAndTrustStores(File baseDir, TlsConfig tlsConfig, NiFiPropertiesWriterFactory niFiPropertiesWriterFactory, List<String> hostnames, List<String> keyStorePasswords,
+                                                  List<String> keyPasswords, List<String> trustStorePasswords, String httpsPort) throws GeneralSecurityException, IOException {
+        String signingAlgorithm = tlsConfig.getSigningAlgorithm();
+        int days = tlsConfig.getDays();
+        String keyPairAlgorithm = tlsConfig.getKeyPairAlgorithm();
+        int keySize = tlsConfig.getKeySize();
+        TlsCertificateAuthorityManager tlsCertificateAuthorityManager = new TlsCertificateAuthorityManager(tlsConfig);
+        KeyStore.PrivateKeyEntry privateKeyEntry = tlsCertificateAuthorityManager.getOrGenerateCertificateAuthority();
+        X509Certificate certificate = (X509Certificate) privateKeyEntry.getCertificateChain()[0];
+        KeyPair caKeyPair = new KeyPair(certificate.getPublicKey(), privateKeyEntry.getPrivateKey());
+
+        try (PemWriter pemWriter = new PemWriter(new OutputStreamWriter(outputStreamFactory.create(new File(baseDir, ROOT_CERT_CRT))))) {
+            pemWriter.writeObject(new JcaMiscPEMGenerator(certificate));
+        }
+
+        try (PemWriter pemWriter = new PemWriter(new OutputStreamWriter(outputStreamFactory.create(new File(baseDir, ROOT_CERT_PRIVATE_KEY))))) {
+            pemWriter.writeObject(new JcaMiscPEMGenerator(caKeyPair));
+        }
+
+        for (int i = 0; i < hostnames.size(); i++) {
+            String hostname = hostnames.get(i);
+            File hostDir = new File(baseDir, hostname);
+
+            if (!hostDir.mkdirs()) {
```

Comment:
If the directory already exists, this will return `false` and throw an exception. Check if the directory exists and either log a warning before clearing it, or skip generating that hostname's directory. 

Example:

```bash
hw12203:...assembly/target/nifi-toolkit-1.0.0-SNAPSHOT-bin/nifi-toolkit-1.0.0-SNAPSHOT (pr695) alopresto
🔓 355s @ 15:34:06 $ ./bin/tls-toolkit.sh standalone -n host1, host2
hw12203:...assembly/target/nifi-toolkit-1.0.0-SNAPSHOT-bin/nifi-toolkit-1.0.0-SNAPSHOT (pr695) alopresto
🔓 15s @ 15:34:22 $ tree
.
├── host1
│   ├── keystore.jks
│   ├── nifi.properties
│   └── truststore.jks
├── localhost
│   ├── keystore.jks
│   ├── nifi.properties
│   └── truststore.jks
├── rootCert.crt
└── rootCert.key

5 directories, 37 files
hw12203:...assembly/target/nifi-toolkit-1.0.0-SNAPSHOT-bin/nifi-toolkit-1.0.0-SNAPSHOT (pr695) alopresto
🔓 65s @ 15:35:31 $ ./bin/tls-toolkit.sh standalone -n host1,host2
Error creating generating tls configuration. (Unable to make directory: /Users/alopresto/Workspace/nifi/nifi-toolkit/nifi-toolkit-assembly/target/nifi-toolkit-1.0.0-SNAPSHOT-bin/nifi-toolkit-1.0.0-SNAPSHOT/./host1)
```